### PR TITLE
Prevent duplicated UUUIDs in inner blocks when copying container blocks

### PR DESCRIPTION
**Pull request by sneridagh**
----
📚 Documentation preview 📚: https://volto--6112.org.readthedocs.build/

### CODE REVIEW

File: packages/volto/src/helpers/Blocks/cloneBlocks.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      blocks: {
+        a: { '@type': 'slate', value: 1 },
+        b: { '@type': 'slate', value: 2 },
+      },
+      blocks_layout: { items: ['a', 'b'] },
+    };
+    const result = cloneBlocks(blocksData);
+
+    expect(result.blocks_layout.items[0]).not.toBe('a');
+    expect(result.blocks[result.blocks_layout.items[0]]).not.toBe({
```
Comment (davisagli) [inline packages/volto/src/helpers/Blocks/cloneBlocks.test.js:16]: Isn't this the assertion we want here?

```suggestion
    expect(result.blocks[result.blocks_layout.items[0]]).toStrictEqual({
```

File: packages/volto/src/components/manage/Form/BlocksToolbar.jsx [hunk 2]
```diff
@@ -97,8 +98,8 @@ export class BlocksToolbarComponent extends React.Component {
         return mode === 'copy'
           ? blockConfig.cloneData
             ? blockConfig.cloneData(blockData)
-            : [uuid(), blockData]
-          : [uuid(), blockData]; // if cut/pasting blocks, we don't clone
+            : [uuid(), cloneBlocks(blockData)]
+          : [uuid(), cloneBlocks(blockData)]; // if cut/pasting blocks, we don't clone
       })
       .filter((info) => !!info); // some blocks may refuse to be copied
```
Comment (davisagli) [inline packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: Can we avoid duplicating cloneWithIds between here and cloneBlocks.ts?

Comment (sneridagh) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: @davisagli do you mean unify all in a single helper? or improve the condition?

File: packages/volto/src/helpers/Blocks/cloneBlocks.test.js (hunk 1 above)
Comment (sneridagh) [reply packages/volto/src/helpers/Blocks/cloneBlocks.test.js]: Fixed

File: packages/volto/src/components/manage/Form/BlocksToolbar.jsx (hunk 2 above)
Comment (davisagli) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: @sneridagh I mean use a helper function in both places

Comment (sneridagh) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: @davisagli let's talk before you leave, I think that the condition was wrong, from the beginning. If it's not copy, does the same as copy, maybe it's a copy/paste omission error? I'd say if it's move, we should not clone.

Comment (davisagli) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: @sneridagh Ah I see what you are saying.

You can ignore my comment here, I thought this was the same code as in cloneBlocks.ts but it isn't.

But you're right, cloneBlocks should not be used when moving

Comment (sneridagh) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:104]: @davisagli okay, I implemented a way to preserve the original uuid if it's a cut operation.

File: packages/volto/src/components/manage/Form/BlocksToolbar.jsx [hunk 3]
```diff
@@ -91,14 +92,14 @@ export class BlocksToolbarComponent extends React.Component {
...
+      .filter(([blockId, blockData]) => blockId && !!blockData['@type']) // Removes null blocks
+      .map(([blockId, blockData]) => {
         const blockConfig = config.blocks.blocksConfig[blockData['@type']];
         return mode === 'copy'
           ? blockConfig.cloneData
             ? blockConfig.cloneData(blockData)
-            : [uuid(), blockData]
-          : [uuid(), blockData]; // if cut/pasting blocks, we don't clone
+            : [uuid(), cloneBlocks(blockData)]
+          : [blockId, cloneBlocks(blockData)]; // if cut/pasting blocks, we don't clone
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx:102]: ummm correct!

[the remark this answers is not in the document]
Comment (sneridagh) [reply packages/volto/src/components/manage/Form/BlocksToolbar.jsx]: Fixed.

**Review by davisagli — APPROVED** [review]
@sneridagh That looks better now

**Pull request merged**